mdb/messages.py

```diff
@@ -0,0 +1,111 @@
+# Copyright 2023-2024 Tom Meltzer. See the top-level COPYRIGHT file for
+# details.
+
+import json
+from dataclasses import dataclass
+from typing import Any
+
+MDB_CLIENT = "mdb client"
+DEBUG_CLIENT = "debug client"
+EXCHANGE = "exchange server"
+
+END_BYTES = b"_MDB_END_"
+
+
+@dataclass
+class Message:
+    msg_type: str
+    data: dict[str, Any]
+
+    @staticmethod
+    def from_json(text: bytes) -> "Message":
+        msg = json.loads(text.decode()[: -len(END_BYTES)])
+        return Message(msg["msg_type"], msg["data"])
+
+    @staticmethod
+    def debug_conn_request() -> "Message":
+        return Message(
+            "debug conn request",
```

Comment:
This is just my personal preference, but I would recommend avoiding spaces in the message types, and use either hierarchical `"debug.conn.request"`, else snake case `"debug_conn_request"` as it makes finding them and filtering in the logs a bit easier.

Reply:
good idea :bulb: 

I decided on snake case ... because python :snake: